sdk/search/azure-search/src/main/java/com/azure/search/util/AutocompletePagedFlux.java

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.search.util;
+
+import com.azure.core.http.rest.PagedFluxBase;
+import com.azure.search.AutocompletePagedResponse;
+import com.azure.search.models.AutocompleteItem;
+import reactor.core.publisher.Mono;
+
+import java.util.function.Supplier;
+
+/**
+ * Implementation of {@link PagedFluxBase} where the element type is {@link AutocompleteItem} and the page type is
+ * {@link AutocompletePagedResponse}.
+ */
+public final class AutocompletePagedFlux extends PagedFluxBase<AutocompleteItem, AutocompletePagedResponse> {
```

Comment:
I did not understand why we still use  PagedFluxBase here. Do we want to deprecate it?

Reply:
The deprecation of PagedFluxBase is still under discussion, I believe we are leaning towards not deprecating it as one of the most common scenarios for specialized `PagedFlux`s is what is covered by `PagedFluxBase`.

@JonathanGiles @anuchandy @srnagar

Reply:
That's right, since PagedFluxBase is still a common type for some services where continuation token is string and the response is a type of `PagedResponse`, we don't want to eagerly deprecate something that might be useful.